src/modules/0.0.19/aws_ecs_simplified/sql/create_triggers.sql

```diff
@@ -0,0 +1,320 @@
+-- ##############################
+
+CREATE OR REPLACE FUNCTION sync_ecs_simplified()
+RETURNS TRIGGER LANGUAGE plpgsql AS $$
+DECLARE
+  is_valid BOOLEAN;
+  _app_name TEXT;
+  _app_port INTEGER;
+  _target_group_name TEXT;
+  _cluster_name TEXT;
+  _desired_count INTEGER;
+  _task_definition_id INTEGER;
+  _image_tag TEXT;
+  _image_digest TEXT;
+  _public_ip BOOLEAN;
+  _repository_uri TEXT;
+  _log_group_name TEXT;
+  _load_balancer_name TEXT;
+  _repository_name TEXT;
+  _load_balancer_dns TEXT;
+  _service_name TEXT;
+  _env_variables TEXT;
+  _force_new_deployment BOOLEAN;
+  _cpu_mem TEXT;
+  _cpu INTEGER;
+  _mem INTEGER;
+  _security_group_id INTEGER;
+BEGIN
+  -- clear out the ecs_simplified table and completely recreate it so there is no logic specific to the table / trigger input and so that deletes work since this is an AFTER UPDATE/INSERT/DELETE trigger
+  DELETE FROM ecs_simplified;
+  FOR _service_name, _desired_count, _public_ip, _cluster_name, _target_group_name, _task_definition_id, _force_new_deployment IN
+  SELECT name, desired_count, assign_public_ip = 'ENABLED', cluster_name, target_group_name, task_definition_id, force_new_deployment FROM service
+  LOOP
+
+    _app_name = SPLIT_PART(_service_name, '-', 1);
+
+    SELECT port INTO _app_port
+    FROM target_group
+    WHERE target_group_name = _target_group_name AND target_type = 'ip' AND protocol = 'HTTP' AND health_check_path = '/health';
+
+    SELECT security_group_id INTO _security_group_id
+    FROM service_security_groups
+    WHERE service_name = _service_name LIMIT 1;
+
+    is_valid = _app_name IS NOT NULL AND _app_port IS NOT NULL AND _security_group_id IS NOT NULL;
+
+    is_valid = is_valid AND 1 = (SELECT COUNT(*) FROM security_group_rule WHERE security_group_id = _security_group_id AND is_egress = false AND ip_protocol = 'tcp' AND from_port = _app_port AND to_port = _app_port AND cidr_ipv4 = '0.0.0.0/0');
+
+    is_valid = is_valid AND 1 = (SELECT COUNT(*) FROM security_group_rule WHERE security_group_id = _security_group_id AND is_egress = true AND ip_protocol = '-1' AND from_port = -1 AND to_port = -1 AND cidr_ipv4 = '0.0.0.0/0');
+
+    SELECT cpu_memory::TEXT
+    INTO _cpu_mem
+    FROM task_definition
+    WHERE id = _task_definition_id AND execution_role_name = task_role_name;
+
+    -- TODO get latest or sort ?
+    SELECT tag, digest, repository_name, log_group_name, cpu, memory, image, env_variables
+    INTO _image_tag, _image_digest, _repository_name, _log_group_name, _cpu, _mem, _repository_uri, _env_variables
+    FROM container_definition
+    WHERE task_definition_id = _task_definition_id AND host_port = _app_port AND container_port = _app_port AND essential = true LIMIT 1;
+
+    IF _cpu_mem IS NULL THEN
+      _cpu_mem = get_cpu_mem_enum_from_parts(_cpu, _mem);
+    END IF;
+
+    IF _repository_name IS NOT NULL THEN
+      SELECT repository_uri INTO _repository_uri FROM repository WHERE repository_name = _repository_name;
+    END IF;
+
```

Comment:
should not be here an `is_valid` check for the props you just have checked (cpu_mem and/or repository_name)?

Reply:
well, both parameters can be null/optional until they hit their respective conditionals. or do you mean adding a new check after the conditionals? I will do that

Reply:
actually, `_repository_name` *can* be NULL so that broke the test. reverted back to what I had